test/chart/setDomain.js

```diff
@@ -0,0 +1,106 @@
+        describe('setDomain() is called for a chart with two linear axes', () => {
+            ['x','y'].forEach(axis => {
+                it('returns a non-zero-length array for the specified axis', () => {
+                    console.log(`    - testing ${charts.linear_linear[axis].type} ${axis} axis:`);
+                    expect(charts.linear_linear[axis + '_dom'].length).toBeGreaterThan(0);
+                });
+
+                it('does not return a zero-range domain for the specified axis', () => {
+                    const test_data = [];
+                    data.forEach(d => {
+                        const datum = {};
+                        for (const prop in d)
+                            datum[prop] = d[prop];
+                        datum[charts.linear_linear[axis].column] = 0;
+                        test_data.push(datum);
+                    });
+                    charts.linear_linear.consolidateData(test_data);
+                    expect(charts.linear_linear[axis + '_dom'][1] - charts.linear_linear[axis + '_dom'][0]).toBeGreaterThan(0);
+                });
+
+                it('gives precedence to the domain defined in the config', () => {
+                    const domain = [-100,100];
+                    charts.linear_linear.config[axis].domain = domain.slice();
+                    charts.linear_linear.consolidateData(data);
+                    expect(charts.linear_linear[axis + '_dom'].join(',')).toEqual(domain.join(','));
+                });
+            });
+        });
+
+        //ordinal x linear
+        describe('setDomain() is called for a chart with an ordinal x-axis and a linear y-axis', () => {
+            it('returns a unique set of the ordinal axis variable as its domain', () => {
+                const ordinalDomain = d3
+                    .set(data.map(d => d[charts.ordinal_linear.config.x.column]))
+                    .values()
+                    .sort()
+                    .join(', ');
+                expect(charts.ordinal_linear.x_dom.sort().join(', ')).toEqual(ordinalDomain);
+            });
+
+            it('sets the lower limit of the linear domain to 0', () => {
+                expect(charts.ordinal_linear.y_dom[0]).toEqual(0);
+            });
+
+            it('sets the upper limit of the linear domain to the maximum value of the linear axis variable', () => {
+                const upperLimit = d3.max(data.map(d => d[charts.ordinal_linear.config.y.column]));
+                expect(charts.ordinal_linear.y_dom[1]).toEqual(upperLimit);
+            });
+        });
+
+        //linear x ordinal
+        describe('setDomain() is called for a chart with an linear x-axis and a ordinal y-axis', () => {
+            it('sets the lower limit of the linear domain to 0', () => {
+                expect(charts.linear_ordinal.x_dom[0]).toEqual(0);
+            });
+
+            it('sets the upper limit of the linear domain to the maximum value of the linear axis variable', () => {
```

Comment:
Is it worth adding a test where `behavior` isn't set to `flex` and checking that it is still possible to have a non-zero lower limit for bars?

Reply:
So like, verify that without a user-defined domain and with `behavior` unspecified or set to `raw` that the lower limit is 0?

Reply:
I was actually thinking that we want to confirm that *with* a user defined domain (e.g. [10, 20]) and behavior 'flex' the lower limit is *not* 0. (e.g. 10)

Reply:
https://github.com/RhoInc/Webcharts/issues/202